### ISSUE TITLE
Slim down docker image

### ISSUE THREAD

**Issue by cjh1** [girder/girder#2403]
Look at using alpine as base image as well as using "FROM as" syntax for build step.

**Comment by kotfic** [comment]
Keep in mind that some downstream projects build on girder/girder:latest.  A change of this magnitude might be considered "breaking"

**Comment by cjh1** [comment]
Yes, we should probably do this for Girder 3

**Comment by danlamanna** [comment]
I added some of my Alpine experimentations to [this branch](https://github.com/girder/girder/compare/slimify-docker-images), running our regular Dockerfile yields an image that's 1.68GB, the Alpine version is 1.02GB.

**Comment by mgrauer** [comment]
We could achieve this by adding an additional lighter weight Docker container based on e.g. Alpine, or using a multi-stage build (though that wouldn't work for downstreams building Girder plugins). Other related issues are that all plugins are built by default and some have huge dependency trees.

**Comment by subdavis** [comment]
``` dockerfile
FROM python:3.7-slim as runtime
EXPOSE 8080

WORKDIR /girder

RUN apt update && apt install -qy \
  gcc \
  git
COPY . /girder/

# TODO: Do we want to create editable installs of plugins as well?  We
# will need a plugin only requirements file for this.
RUN pip install --upgrade --upgrade-strategy eager .

COPY --from=girder/girder:latest /usr/share/girder/static/ /usr/local/share/girder/static/

ENTRYPOINT ["girder", "serve"]
```

This slims us down to 534M.  However, we can shave off another 200M-240M by dropping the production dependency on `git` because when building for production, `setuptools-scm` isn't even being used.

I don't know how to avoid installing `setup_requires` dependencies with pip or how to best rework setup.py, but I think it's worth it to get the image down into the 200 range.

**Comment by brianhelba** [comment]
> I don't know how to avoid installing `setup_requires` dependencies with pip or how to best rework setup.py, but I think it's worth it to get the image down into the 200 range.

@subdavis Agreed. I have some crazy ideas about adding environment variable or config file overrides for the automatic version detection, but they're not fleshed out. I'd be happy to discuss this when you have time.

**Comment by manthey** [comment]
One way to do this is with a single command in the Docker that removes git after use:
```
COPY . /girder/
RUN apt-get update \
  && apt-get install -qy --no-install-recommends \
    git \
 && pip install \
  --no-cache-dir \
  --find-links https://girder.github.io/large_image_wheels \
  && apt-get remove -y git \
  && apt-get autoremove -y \
  && rm -rf /var/lib/apt/lists/*
```

**Comment by brianhelba** [comment]
> One way to do this is with a single command in the Docker that removes git after use:
> 
> ```
> COPY . /girder/
> RUN apt-get update \
>   && apt-get install -qy --no-install-recommends \
>     git \
>  && pip install \
>   --no-cache-dir \
>   --find-links https://girder.github.io/large_image_wheels \
>   && apt-get remove -y git \
>   && apt-get autoremove -y \
>   && rm -rf /var/lib/apt/lists/*
> ```

Can we do a multi-multi-stage build, where we install all Python packages in a build container (with Git, GCC, etc.), then copy over the `site-packages`?
